stock_getter/backend_stockgetter/stock_service/forecast_tools/tests_forecast_tools/test_ProphetModel.py

```diff
@@ -0,0 +1,97 @@
+###################################################################
+# File for testing Forecaster class
+###################################################################
+import unittest
+from ..forecaster import ProphetModel
+import pandas as pd
+
+
+###################################################################
+# Class implementing Forecaster tests
+###################################################################
+class TestProphetModel(unittest.TestCase):
+    TURN_OFF_ALL_TESTS = False
+
+    def setUp(self) -> None:
+        """This runs before every test and sets up needed objects
+        and artifacts for testing
+        """
+        # Load csv file with stock data for testing
+        fp = './backend_stockgetter/stock_service/forecast_tools'
+        fp += '/tests_forecast_tools/test_data/aapl_stk.csv'
+        self.stock_data = pd.read_csv(fp, parse_dates=['date'])
+        self.grain = 'daily'
+        self.horizon = 30
+        print('Starting tests for ProphetModel object')
+
+    def tearDown(self) -> None:
+        """Terdown function
+        for now, we just call the super method. No custom logic
+        here"""
+        return super().tearDown()
+
+
+###################################################################
+# Unit tests
+###################################################################
+    def test_run_forecast(self):
+        """Test run_forecast function
+        """
+        # Instantiate Prophet model object and run forecast
+        forecast_model = ProphetModel(self.stock_data, self.grain)
+        fcast_df = forecast_model.run_forecast(self.horizon)
+
+        # Run checks here
+        msg = "The return object should be a pandas data frame"
+        self.assertTrue(isinstance(fcast_df, pd.DataFrame))
+        
+        msg = "Number of elements in the data frame should be 5715"
+        self.assertEqual(len(fcast_df), 5715)
+
+        msg = "'date' should be one of the data frame columns"
+        self.assertTrue('date' in fcast_df.columns)
```

Comment:
We seem to be repeating this assertion with only change being the column name. We can extract all of these to separate test and use a loop or we can use parametrized feature similar to the PR I have for unit tests

Reply:
I disagree. This looks fine as is and is clear and readable. Turning it into the loop makes this worse.

Reply:
I am not sure if you think I was only talking about line 46 to 52. I was referring to line 52 to line 82, bad smell with unnecessary repetitions. Extracting and using parametrized would be a better solution.

Reply:
Parametrizing this makes it harder to maintain. I like this as is.

Reply:
I am ok with leaving this as is, but would still like to understand for my own benefit and make sure I am not missing something. Can you point to a part that is difficult to read and hard to maintain in a code like below that can be used to replace what you have from line 52 to 82:
![image](https://user-images.githubusercontent.com/95208662/172237352-2ffa7d10-9fbd-4f06-9521-3bb389ad8a0c.png)

(Not sure what msg is used for but included it the way it is now, and was unable to format nicer using the github code formatter, so putting as image here)

Can you think of a scenario that makes this difficult to maintain? Scenarios like adding/removing a column are  straightforward, and adding additional check for each of the columns would be a single change in the test body instead of making multiple similar changes.